src/deploy/functions/args.ts

```diff
@@ -5,12 +5,24 @@ import * as deployHelper from "./functionsDeployHelper";
 
 // These types should probably be in a root deploy.ts, but we can only boil the ocean one bit at a time.
 
+interface CodebasePayload {
+  wantBackend: backend.Backend;
+  haveBackend: backend.Backend;
+}
+
 // Payload holds the output of what we want to build + what we already have.
 export interface Payload {
-  functions?: {
-    wantBackend: backend.Backend;
-    haveBackend: backend.Backend;
-  };
+  codebase?: CodebasePayload;
```

Comment:
This is a large object that is passed around for all deployment targets. It's still probably appropriate to have them in side a "functions" key. Also, are we not going to have an array of codebases or map of them soon?

Reply:
Yes exactly. I'm trying to prepare the stage so that adding codebase -> <whatever> mapping seems more obvious.